File: daq/slc/src/dqm/src/Range.cc
#include "dqm/Range.h"

using namespace Belle2;

Range::Range() : Histo1D("", "", 2, 0, 1) {}

Range::Range(const std::string& name) : Histo1D(name, name, 2, 0, 1) {}

Range::Range(const std::string& name, double min, double max) : Histo1D(name, name, 2, 0, 1)
{
  _data->set(0, min);
  _data->set(1, max);
}

Range::~Range() throw() {}

void Range::set(double min, double max) throw()
{
  _data->set(0, min);
  _data->set(1, max);
}

void Range::reset() throw() {}

void Range::fixMaximum(double data, bool fix) throw()
{
  _data->set(1, data);
}

void Range::fixMinimum(double data, bool fix) throw()
{
  _data->set(0, data);
}

int Range::getDim() const throw()
{
  return 0;
}

double Range::getMaximum() const throw()
{
  return _data->get(1);
}

double Range::getMinimum() const throw()
{
  return _data->get(0);
}

void Range::setMaximum(double data) throw()
{
  _data->set(1, data);
}

void Range::setMinimum(double data) throw()
{
  _data->set(0, data);
}

void Range::readConfig(Belle2::Reader& reader) throw(Belle2::IOException)
{
  MonObject::readConfig(reader);
  setMinimum(reader.readDouble());
  setMaximum(reader.readDouble());
}

void Range::writeConfig(Belle2::Writer& writer) const throw(Belle2::IOException)
{
  MonObject::writeConfig(writer);
  writer.writeDouble(getMinimum());
  writer.writeDouble(getMaximum());
}

void Range::readUpdate(Belle2::Reader& reader) throw(Belle2::IOException)
{

}

void Range::writeUpdate(Belle2::Writer& writer) const throw(Belle2::IOException)
{

}

void Range::readUpdateFull(Belle2::Reader& reader) throw(Belle2::IOException)
{

}

void Range::writeUpdateFull(Belle2::Writer& writer) const throw(Belle2::IOException)
{

}

File: daq/slc/extra/javalib/src/b2daq/dqm/core/Range.java
package b2daq.dqm.core;

import java.io.IOException;

import b2daq.core.Reader;
import b2daq.core.Writer;

public abstract class Range extends Histo {
		
	public Range (String name, double min, double max) {
		super(name, name);
		_data_v = new DoubleArray(2);
		_data_v.set(0, min);
		_data_v.set(1, max);
	}
	
	public void set(double min, double max) {
		_data_v.set(0, min);
		_data_v.set(1, max);
	}
	
	@Override
	public void fixMaximum(double data, boolean fix) {
		setMaximum(data);
	}

	@Override
	public void fixMinimum(double data, boolean fix) {
		setMinimum(data);
	}

	public void reset() {}

	@Override
	public int getDim() {
		return 0;
	}

	@Override
	public double getMaximum() {
		return _data_v.get(1);
	}

	@Override
	public double getMinimum() {
		return _data_v.get(0);
	}

	@Override
	public void setMaxAndMin() {}

	@Override
	public void setMaximum(double data) {
		_data_v.set(1, data);
	}

	@Override
	public void setMinimum(double data) {
		_data_v.set(0, data);
	}

	@Override
	public void readConfig(Reader reader) throws IOException {
		setTabId(reader.readChar());
		setPositionId(reader.readChar());
		setName(reader.readString());
		setMinimum(reader.readDouble());
		setMaximum(reader.readDouble());
	}

	@Override
	public void writeConfig(Writer writer) throws IOException {
		writer.writeString(getDataType());
		writer.writeChar((char)getTabId());
		writer.writeChar((char)getPositionId());
		writer.writeString(getName());
		writer.writeDouble(getMinimum());
		writer.writeDouble(getMaximum());
	}

	public void readUpdate(Reader reader) throws IOException {}
	public void writeUpdate(Writer writer) throws IOException {}

}
